File: JAVA/14thOct.java
class max_diff{ 

	static int maxSortedAdjacentDiff(int[] arr, int n) { 
		int maxVal = arr[0]; 
		int minVal = arr[0]; 
		for (int i = 1; i < n; i++) { 
			maxVal = Math.max(maxVal, arr[i]); 
			minVal = Math.min(minVal, arr[i]); 
		}  
		int maxBucket[] = new int[n - 1]; 
		int minBucket[] = new int[n - 1]; 
		Arrays.fill(maxBucket, 0, n - 1, Integer.MIN_VALUE); 
		Arrays.fill(minBucket, 0, n - 1, Integer.MAX_VALUE); 

		float delta = (float) (maxVal - minVal) / (float) (n - 1);  
		for (int i = 0; i < n; i++) { 
			if (arr[i] == maxVal || arr[i] == minVal) { 
				continue; 
			} 
			int index = (int) (Math.round((arr[i] - minVal) / delta)); 
			if (maxBucket[index] == Integer.MIN_VALUE) { 
				maxBucket[index] = arr[i]; 
			} else { 
				maxBucket[index] = Math.max(maxBucket[index], arr[i]); 
			} 
			if (minBucket[index] == Integer.MAX_VALUE) { 
				minBucket[index] = arr[i]; 
			} else { 
				minBucket[index] = Math.min(minBucket[index], arr[i]); 
			} 
		} 
		int prev_val = minVal; 
		int max_gap = 0; 
		for (int i = 0; i < n - 1; i++) { 
			if (minBucket[i] == Integer.MAX_VALUE) { 
				continue; 
			} 
			max_gap = Math.max(max_gap, minBucket[i] - prev_val); 
			prev_val = maxBucket[i]; 
		} 
		max_gap = Math.max(max_gap, maxVal - prev_val); 

		return max_gap; 
	} 

File: JavaScript/14thOct.js
let sample_input_arr1 = [1,7,0,9,12];
let sample_input_arr2 = [9,7,0,2,12,1,17];

function maxConsecutiveGap(...arr){
    n = arr.length;
    for(var k=0;k<n;k++){
        for(var j=0;j<n-k-1;j++){
            if(arr[j+1]<arr[j]){
                temp = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = temp;
            }
        }
    }
    sorted_arr = arr;
    max_diff = -1;
    for(var i=0;i<sorted_arr.length;i++){
        if(Math.abs(sorted_arr[i+1]-sorted_arr[i])>max_diff){
            max_diff = Math.abs(sorted_arr[i+1]-sorted_arr[i]);
        }
    }
    return max_diff;
}

console.log(maxConsecutiveGap(...sample_input_arr1));
console.log(maxConsecutiveGap(...sample_input_arr2));